packages/store/tests/inheritance.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { TestBed } from '@angular/core/testing';
+
+import { Store } from '../src/store';
+import { State } from '../src/decorators/state';
+import { NgxsModule } from '../src/module';
+
+describe('Inheritance @State', () => {
+  interface MyStateModel {
+    value: string;
+  }
+
+  @State<string>({
+    name: 'child_a',
+    defaults: 'child_a'
+  })
+  class MyChildAState {}
+
+  @State<string>({
+    name: 'child_b',
+    defaults: 'child_b'
+  })
+  class MyChildBState {}
+
+  @State<MyStateModel>({
+    name: 'a',
+    defaults: { value: 'a' },
+    children: [MyChildAState]
+  })
+  class MyState {}
+
+  @State<MyStateModel>({
+    name: 'b',
+    children: [MyChildBState]
+  })
+  class MyOtherState extends MyState {}
```

Comment:
I guess for the inherited children scenario we would have a test that has another variant of this inherited state that does not override the children. The test would essentially just be defining how ngxs currently handles this for now (an error thrown or just silently accept it) - a characterization test.